MetaMask/metamask-extension
# split advancedGasFee by network and erase previous options

## Explanation
Splits out the advancedGasFee setting to support a per-network setting keyed by chainId. 

Fixes #13800
FIxes #20573

## Manual Testing Steps
1. Create a transaction using the send flow.
2. On the confirmation screen click "Market" (or whatever it says in the attached screenshot)
<img width="355" alt="Screenshot 2023-08-23 at 3 44 25 PM" src="https://github.com/MetaMask/metamask-extension/assets/4448075/e818e13c-3416-4188-beb6-8901f5171852">

3. Click the "advanced" option
<img width="293" alt="Screenshot 2023-08-23 at 3 45 20 PM" src="https://github.com/MetaMask/metamask-extension/assets/4448075/cf30d378-26fb-4fd6-b452-630185a17880">

4. Set a higher priority and maxFee 
<img width="312" alt="Screenshot 2023-08-23 at 3 45 27 PM" src="https://github.com/MetaMask/metamask-extension/assets/4448075/4f534237-da0f-4791-bcc2-7a3dcbcda6d3">

5. Click the checkbox to use these by default. 
6. save options.
7. cancel the confirmation.

Repeat the process on the same network and see that instead of market it says "advanced" and if you click on it you can see your settings applied. 

Switch network and repeat the process and see it says "market" and does not have your advanced settings. Set a DIFFERENT set of values as your advanced setting and save it. Now switch between the two networks and see that your options persist on transactions created on that network but are not applied to other networks or each other. 

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
